airflow/serialization/helpers.py

```diff
@@ -38,7 +39,19 @@ def is_jsonable(x):
         else:
             return True
 
+    max_size = conf.getint("core", "max_templated_field_size")
     if not is_jsonable(template_field):
-        return str(template_field)
+        serialized = str(template_field)
+        if len(serialized) > max_size:
+            return (
+                "Value removed due to size. "
+                "You can change this behaviour in [core]max_templated_field_size"
+            )
+        return serialized
     else:
+        if template_field and len(str(template_field)) > max_size:
+            return (
+                "Value removed due to size. "
+                "You can change this behaviour in [core]max_templated_field_size"
+            )
```

Comment:
Since you are always comparing the str length, you don't have to do it in each branch, you can do it before the if.

Reply:
Changing to doing it in separate branches cause we can't use `if` statement on dataframe. It gives a valueerror